featuretests/bakerystorage_test.go

```diff
@@ -87,30 +85,36 @@ func (s *BakeryStorageSuite) ensureIndexes(c *gc.C) {
 }
 
 func (s *BakeryStorageSuite) TestCheckNewMacaroon(c *gc.C) {
-	mac, err := s.service.NewMacaroon(nil)
+	cav := []checkers.Caveat{{Condition: "something"}}
+	mac, err := s.bakery.Oven.NewMacaroon(context.TODO(), bakery.LatestVersion, cav, bakery.NoOp)
 	c.Assert(err, jc.ErrorIsNil)
-	_, err = s.service.CheckAny([]macaroon.Slice{{mac}}, nil, nil)
+	_, _, err = s.bakery.Oven.VerifyMacaroon(context.TODO(), macaroon.Slice{mac.M()})
 	c.Assert(err, gc.ErrorMatches, "verification failed: macaroon not found in storage")
 
 	store := s.store.ExpireAfter(10 * time.Second)
-	mac, err = s.service.WithStore(store).NewMacaroon(nil)
+	b := bakery.New(bakery.BakeryParams{
+		RootKeyStore: store,
+	})
+	mac, err = b.Oven.NewMacaroon(context.TODO(), bakery.LatestVersion, cav, bakery.NoOp)
 	c.Assert(err, jc.ErrorIsNil)
-	_, err = s.service.CheckAny([]macaroon.Slice{{mac}}, nil, nil)
+	op, conditions, err := s.bakery.Oven.VerifyMacaroon(context.TODO(), macaroon.Slice{mac.M()})
 	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(op, jc.DeepEquals, []bakery.Op{bakery.NoOp})
+	c.Assert(conditions, jc.DeepEquals, []string{"something"})
 }
 
 func (s *BakeryStorageSuite) TestExpiryTime(c *gc.C) {
 	// Reinitialise bakery service with storage that will expire
 	// items immediately.
 	s.initService(c, true)
 
-	mac, err := s.service.NewMacaroon(nil)
+	mac, err := s.bakery.Oven.NewMacaroon(context.TODO(), bakery.LatestVersion, nil, bakery.NoOp)
 	c.Assert(err, jc.ErrorIsNil)
 
 	// The background thread that removes records runs every 60s.
 	// Give a little bit of leeway for loaded systems.
 	for i := 0; i < 90; i++ {
-		_, err = s.service.CheckAny([]macaroon.Slice{{mac}}, nil, nil)
+		_, _, err = s.bakery.Oven.VerifyMacaroon(context.TODO(), macaroon.Slice{mac.M()})
 		if err == nil {
 			time.Sleep(time.Second)
```

Comment:
not related to current pr, but not sure if we can inject clock and mock for testing.

Reply:
sadly the clock is hard wired in the upstream repo